### PR TITLE
Refactor: sync useLocalStorage across components and tabs

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

Although the localStorage is global and was shared across all calls of the `useLocalStorage` hook, it wasn't triggering an update when one of the instances sets a new value.

This PR subscribes to the global `storage` event and syncs the hooks across components and browser tabs.

### CODE REVIEW

File: src/services/local-storage/useLocalStorage.ts
Comment (usame-algan) [inline src/services/local-storage/useLocalStorage.ts:43]: Do we need to check anything here or can we simply return `newValue`?

For example if we use the hook twice for the same local storage key and one hook updates its value to `undefined`, should the other hook not also update its cache to `undefined`?

Comment (katspaugh) [reply src/services/local-storage/useLocalStorage.ts:43]: The current usage of this hook unfortunately expects that the value is never undefined. I think it would be more correct if you could set it to undefined, but I kept the previous behavior.

Comment (usame-algan) [reply src/services/local-storage/useLocalStorage.ts]: In safe creation we set the `pendingSafe` to `undefined` when the user cancels the transaction. With the change it is being set to `{}` which causes unintended behaviour. Maybe we could call `local.removeItem` when undefined is passed?

Comment (katspaugh) [reply src/services/local-storage/useLocalStorage.ts]: OK, makes sense, let me allow undefined and remove the initialValue.

File: src/services/local-storage/useLocalStorage.ts
Comment (iamacook) [inline src/services/local-storage/useLocalStorage.ts:43]: Just as a note, I refactored this logic in https://github.com/safe-global/web-core/pull/1001 to happen in `cache` initiation, e.g.

```ts
const [cache, setCache] = useState(() => {
  const value = local.getItem<T>(key)
  return value !== undefined ? lsValue : initialValue
})
```

Comment (katspaugh) [reply src/services/local-storage/useLocalStorage.ts:43]: This’ll cause hydration issues.

Comment (iamacook) [reply src/services/local-storage/useLocalStorage.ts:43]: I see. I'll hold off on adjusting my PR until this has been merged then.

File: src/services/local-storage/useLocalStorage.ts
Comment (usame-algan) [inline src/services/local-storage/useLocalStorage.ts:63]: This seems to do the same as the other useEffect that sets the initial value on mount. Can we remove one or the other?

**Review by iamacook — APPROVED** [review]

File: src/components/create-safe/status/CreationStatus.tsx
Comment (usame-algan) [inline src/components/create-safe/status/CreationStatus.tsx:21]: This is not being used anymore

**Review by usame-algan — APPROVED** [review]
💯

**Pull request merged**
